app/forms/flood_risk_engine/steps/base_address_form.rb

```diff
@@ -0,0 +1,131 @@
+
+        # We assume postcode validation already occured so do not validate the format here
+        @post_code = options[:post_code] if options.key? :post_code
+      end
+
+      # This is a form reader - provides view with addresses for drop down selection
+      def address_list
+        @address_list ||= find_by_postcode
+      end
+
+      # The normal update action is no good for managing addresses, need to wire up continue button to use
+      # a different action
+      def url
+        address_enrollment_step_url(enrollment, enrollment.step, post_code, only_path: true)
+      end
+
+      def validate(params)
+        logger.debug("Validating Address form #{inspect}")
+
+        valid = super(params)
+
+        # Use UPRN to find address data and assign to our local address instance
+
+        valid = assign_via_uprn if valid
+
+        # If validation fails we need to redirect BACK to the address page, supplying the POST CODE
+        # So the drop down can be populated
+        if valid
+          @redirect = false
+        else
+          @redirect = true
+          @redirection_url = address_enrollment_step_url(enrollment, enrollment.step, post_code, only_path: true)
+        end
+
+        valid
+      end
+
+      def save
+        logger.debug("*** SAVE AddressToAssign **** #{address_to_assign.inspect}")
+
+        assign_to_enrollment(address_to_assign) if address_to_assign
+
+        # we only need to redirect (e.g so we can populate address drop-down) on failure,
+        # not on success - since then the journey can continue as normal
+        result = super
+        @redirect = !result
+        result
+      end
+
+      # Derived classes must implement which address instance is being managed
+      def assign_to_enrollment(_address)
+        raise NotImplementedError, "Address form cannnot set underlying Enrollment address"
+      end
+
+      protected
+
+      attr_reader :address_to_assign
+
+      def assign_via_uprn
+        # Use UPRN to find address data
+        address_attributes = find_by_uprn
+
+        if address_attributes
+          # Build and validate the inbound address data against a real Flood Address model
+          logger.debug("Build LocalAuthority address from #{address_attributes.inspect}")
+          @address_to_assign = Address.new(address_attributes)
+
+          return @address_to_assign.validate
+        end
+
+        false
+      end
+
+      def find_by_uprn
+        logger.debug("Start BaseAddressForm Lookup on UPRN #{uprn}")
+
+        finder = FloodRiskEngine::FindAddressByUprn.new(uprn)
+        inbound = finder.search
+
+        # parser used to convert inbound data into a format we can
+        # use to construct an Address
+        parser = FloodRiskEngine::Services::Address::Deserialize::EaFacadeToAddress.new
+
+        flood_address_data = parser.address_data(inbound)
+
+        logger.info("Parsed address #{flood_address_data.first}")
+
```

Comment:
Any `logger.info` calls will be in the production log by default - is that the intention or is this just for visibility in dev/test? I noticed in WEX production logs there is quite a bit of logging noise (some produced by me) and wonder if we should dial things down a bit in RP and avoid `info` unless really useful for live debugging (which is hard to predict I know)? In WEX, have we had to inspect the logs at any point so far, and been grateful for the extra logging output? If not, it might suggest we don't need to log much. Any thoughts?